src/Bicep.Core/Diagnostics/DiagnosticLevel.cs

```diff
@@ -4,6 +4,7 @@ namespace Bicep.Core.Diagnostics
 {
     public enum DiagnosticLevel
```

Comment:
>DiagnosticLevel [](http://example.com/codeflow?start=16&length=15)

It seems we're using this enum for two purposes. Setting a a level on analyzer rules and when emitting diagnostics in the compiler with each having a separate but overlapping set of values. This creates a possibility of emitting diagnositcs with `Off` level in the core compiler.

Can we split it into a separate enums and then convert where appropriate?